util/src/array_serialization.rs

```diff
@@ -0,0 +1,56 @@
+use alloc::vec::Vec;
+use core::convert::TryInto;
+use core::marker::PhantomData;
+
+use serde::de::{SeqAccess, Visitor};
+use serde::ser::SerializeTuple;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+pub fn serialize<S: Serializer, T: Serialize, const N: usize>(
+    data: &[T; N],
+    ser: S,
+) -> Result<S::Ok, S::Error> {
+    let mut s = ser.serialize_tuple(N)?;
+    for item in data {
+        s.serialize_element(item)?;
+    }
+    s.end()
+}
+
+struct ArrayVisitor<T, const N: usize>(PhantomData<T>);
+
+impl<'de, T, const N: usize> Visitor<'de> for ArrayVisitor<T, N>
+where
+    T: Deserialize<'de>,
+{
+    type Value = [T; N];
+
+    fn expecting(&self, formatter: &mut core::fmt::Formatter) -> core::fmt::Result {
+        formatter.write_fmt(format_args!("an array of length {}", N))
+    }
+
+    #[inline]
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let mut data = Vec::with_capacity(N);
```

Comment:
nit: can we use an array upfront rather than doing `try_into()` later?

Reply:
with `MaybeUninit`?